### ISSUE TITLE
[DEPR]: studio-frontend repo

### ISSUE THREAD

**Issue by KristinAoki** [openedx/studio-frontend#381]
_Please first read the parent DEPR ticket, [[DEPR]: All Legacy Studio Frontends](https://github.com/openedx/edx-platform/issues/36275)._

### Proposal Date

2025-02-19

### Target Ticket Acceptance Date

2025-03-05

### First Open edX Named Release Without This Functionality

2025-09-08

Ulmo

### Rationale

This repo is currently only used inside the edX Studio. The React code is imported into Django templates and rendered inside of Studio. The React code is only used on four pages, Assets, Accessibility Form, Course Outline Status, and Checklist. The pages in edX Studio are being update to use React and moved to `frontend-app-course-authoring`. To reduce the redundancy of two studio frontend repos, this one will be removed and the components that are not slated to be redesigned will be copied over to `frontend-app-course-authoring`. 

### Removal

The entire repo will be removed and the following references inside `edx-platform`:

- [accessibility.html](https://github.com/openedx/edx-platform/blob/master/cms/templates/accessibility.html)
- health check, including references from:
  - [checklists.html](https://github.com/openedx/edx-platform/blob/master/cms/templates/checklists.html)
  - [course status](https://github.com/openedx/edx-platform/blob/e2a54e0524147c46262e5251e4649b0e79299afd/cms/templates/course_outline.html#LL238C17-L270C23) in course_outline.html
- legacy files and uploads page
- image upload from the legacy Text editor
  - Removing the image upload is contingent on the legacy Text editor deprecation:  https://github.com/openedx/edx-platform/issues/34692
    - Note that this does _not_ affect the other usages of the tinyMCE editors, eg Announcements.

### Replacement

The majority of the code will be rewritten in `frontend-app-course-authoring`. The accessibility and checklist will be moved to `frontend-app-course-authoring`. The code won't change, but will live in the `frontend-app-course-authoring`.

### Additional Info

Communicated here: https://discuss.openedx.org/t/deprecation-removal-studio-frontend-381/10280

Blocked by https://github.com/openedx/edx-platform/issues/34692

**Comment by kdmccormick** [comment]
I'm excited for this one 🔥

**Comment by dianakhuang** [comment]
@KristinAoki We are in the process of assigning owners for DEPR tickets to ensure that we have a point of contact on the current status of them. I assigned you to this one since you seem to be handling most of the movement of the ticket. Out of curiosity, why is this in 'deprecated' instead of 'removing'? Is it just because we're not doing active work on it?

**Comment by KristinAoki** [comment]
I was not fully sure which it belonged under. The code has been moved to course authoring, but the act of removing the repo  and references to it in `edx-platform` has not begun

**Comment by kdmccormick** [comment]
@KristinAoki , I think Deprecated is the right status. I see you [marked it as deprecated](https://github.com/openedx/studio-frontend/) already, which is great.

How can we move it forward? Ideally, 1 release should go by where both the old pages and the new pages are available for operators to use; has that happened yet? And are the new pages at feature-parity with the old pages?

If it's Yes on both counts, then your team or anyone else in the community would be all-clear to start removing studio-frontend.

**Comment by KristinAoki** [comment]
Redwood will be the first release with the new pages and the old pages. The old pages for the Checklist page and the Accessibility Request page have redirects in edx-plaform that always redirect to the new page. The Files and Uploads page redirect can be toggled with the `contentstore.new_studio_mfe.use_new_files_uploads_page` waffle flag. The new pages are at feature-parity with the old pages.

**Comment by kdmccormick** [comment]
Awesome! Do you have any opposition if Tutor toggles on the new Files & Uploads page by default? We could add opt-out instructions to the Redwood release notes. That would allow removal to begin as early as the day after the Redwood cut date.

**Comment by KristinAoki** [comment]
No opposition.

**Comment by kdmccormick** [comment]
Cool, I will take care of making sure Tutor runs the new page in Redwood, can you update the timeline in the issue description?

**Comment by kdmccormick** [comment]
Searching for the `studiofrontend` tag in edx-platform, I actually see five usages of the repo:

* courseHealthCheck (used by Checklists page)
* courseOutlineHealthStatus (used by legacy Course Outline page)
* assets (used by legacy Files and Uploads page)
* editImageModal (used by legacy Text component editor)
* accessibilityPolicy (used by ??? page)

I'm trying to confirm right now whether all of these pages will be enabled in Redwood and ready to remove in Sumac.

* Course Outline (course_status, course_
* Course Checklists
* Accessibility Policy

**Comment by KristinAoki** [comment]
> Searching for the `studiofrontend` tag in edx-platform, I actually see five usages of the repo:
> 
> * courseHealthCheck (used by Checklists page) ✅ 
When navigating to the Checklist page, users should automatically redirected to {course authoring host}/checklists
> * courseOutlineHealthStatus (used by legacy Course Outline page) ✅ 
> * assets (used by legacy Files and Uploads page) ✅ 
> * editImageModal (used by legacy Text component editor) ✅ 
> * accessibilityPolicy (used by ??? page)
Can be found at{studio host}/accessibility and is moved to {course authoring host}/accessibility

All of the references are not yet deleted in edx-platform

**Comment by kdmccormick** [comment]
> When navigating to the Checklist page, users should automatically redirected to {course authoring host}/checklists

Nice, thanks. Is the MFE page gated by a waffle, or enabled by default?

> Can be found at{studio host}/accessibility and is moved to {course authoring host}/accessibility

Got it, thanks.

**Comment by KristinAoki** [comment]
It is enabled by default because it is just showing information so it was low risk to just redirect by default

**Comment by kdmccormick** [comment]
> It is enabled by default. It is just showing information so it was low risk to just redirect by default

Sounds great @KristinAoki 👍🏻 

Because studio-frontend powers the image uploader for the legacy Text component, it seems like we would need to deprecate the legacy Text editor too. Would you or someone on TNL be able to write that up?

**Comment by KristinAoki** [comment]
Yes, I can create the deprecation notice. Do I create the deprecation ticket in `edx-platform`?

**Comment by kdmccormick** [comment]
Yep. Thanks!

**Comment by KristinAoki** [comment]
Issue for legacy text editor deprecation ind edx-platform, #[34692](https://github.com/openedx/edx-platform/issues/34692)

**Comment by kdmccormick** [comment]
We will archive studio-frontend once all legacy Studio frontends have been removed from edx-platform, so I will just make this DEPR a clean-up item on https://github.com/openedx/edx-platform/issues/36275
